llvm/lib/MCCAS/MCCASObjectV1.cpp

```diff
@@ -3155,25 +3166,153 @@ static AbbrevEntryReader getAbbrevEntryReader(ArrayRef<StringRef> AbbrevEntries,
   return AbbrevEntryReader(AbbrevData);
 }
 
-Error DIEVisitor::visitDIERef(BinaryStreamReader &DataReader,
-                              unsigned AbbrevIdx, StringRef DIEData,
-                              ArrayRef<DIEDataRef> &DIEChildrenStack) {
+static std::pair<bool, uint64_t> getNonULEBFormSize(dwarf::Form Form,
+                                                    dwarf::FormParams FP) {
+  switch (Form) {
+  case dwarf::DW_FORM_addr:
+  case dwarf::DW_FORM_ref_addr: {
+    auto FormSize =
+        (Form == dwarf::DW_FORM_addr) ? FP.AddrSize : FP.getRefAddrByteSize();
+    return std::make_pair(false, FormSize);
+  }
+  case dwarf::DW_FORM_exprloc:
+  case dwarf::DW_FORM_block:
+  case dwarf::DW_FORM_block1:
+  case dwarf::DW_FORM_block2:
+  case dwarf::DW_FORM_block4:
+  case dwarf::DW_FORM_sdata:
+  case dwarf::DW_FORM_udata:
+  case dwarf::DW_FORM_ref_udata:
+  case dwarf::DW_FORM_ref4_cas:
+  case dwarf::DW_FORM_strp_cas:
+  case dwarf::DW_FORM_rnglistx:
+  case dwarf::DW_FORM_loclistx:
+  case dwarf::DW_FORM_GNU_addr_index:
+  case dwarf::DW_FORM_GNU_str_index:
+  case dwarf::DW_FORM_addrx:
+  case dwarf::DW_FORM_strx:
+  case dwarf::DW_FORM_LLVM_addrx_offset:
+  case dwarf::DW_FORM_string:
+  case dwarf::DW_FORM_indirect: {
+    return std::make_pair(true, 0);
+  }
+
+  case dwarf::DW_FORM_implicit_const:
+  case dwarf::DW_FORM_flag_present: {
+    return std::make_pair(false, 0);
+  }
+  case dwarf::DW_FORM_data1:
+  case dwarf::DW_FORM_ref1:
+  case dwarf::DW_FORM_flag:
+  case dwarf::DW_FORM_strx1:
+  case dwarf::DW_FORM_addrx1: {
+    return std::make_pair(false, 1);
+  }
+  case dwarf::DW_FORM_data2:
+  case dwarf::DW_FORM_ref2:
+  case dwarf::DW_FORM_strx2:
+  case dwarf::DW_FORM_addrx2: {
+    return std::make_pair(false, 2);
+  }
+  case dwarf::DW_FORM_strx3: {
+    return std::make_pair(false, 3);
+  }
+  case dwarf::DW_FORM_data4:
+  case dwarf::DW_FORM_ref4:
+  case dwarf::DW_FORM_ref_sup4:
+  case dwarf::DW_FORM_strx4:
+  case dwarf::DW_FORM_addrx4: {
+    return std::make_pair(false, 4);
+  }
+  case dwarf::DW_FORM_ref_sig8:
+  case dwarf::DW_FORM_data8:
+  case dwarf::DW_FORM_ref8:
+  case dwarf::DW_FORM_ref_sup8: {
+    return std::make_pair(false, 8);
+  }
+  case dwarf::DW_FORM_data16: {
+    return std::make_pair(false, 16);
+  }
+
+  case dwarf::DW_FORM_strp:
+  case dwarf::DW_FORM_sec_offset:
+  case dwarf::DW_FORM_GNU_ref_alt:
+  case dwarf::DW_FORM_GNU_strp_alt:
+  case dwarf::DW_FORM_line_strp:
+  case dwarf::DW_FORM_strp_sup: {
+    auto FormSize = FP.getDwarfOffsetByteSize();
+    return std::make_pair(false, FormSize);
+  }
+  case dwarf::DW_FORM_addrx3:
```

Comment:
isn't addrx3's size == 3?

Reply:
Yep my bad!

Reply:
Did you forget to update the patch?

Reply:
Yeah minor mistake on my part, I made the change, but then I changed the function to no longer return a std::pair and rebasing made me lose the addrx3 move, I have fixed that now